doc/upgrade-cluster.md

```diff
@@ -58,6 +58,36 @@ To initiate an upgrade process, a user has to update the `spec.version` field on
 
 After the `version` field is updated, the operator will detect the version change and it will trigger the upgrade process.
 
+### Upgrade guide
+
+> Note: To trigger an upgrade please edit only the fields mentioned in this upgrade guide. We do not recommend clubbing other edits (for performance or scaling or anything else) along with the upgrade trigger. Those can be done either prior to triggering the upgrade or after the upgrade has completed successfully.
+
+#### Upgrades till Pravega 0.6
+
+When upgrading the Pravega Cluster to any version till 0.6, we need to ensure that we do not modify any field other than the `version` field while triggering the upgrade. Altering of any other parameter can be handled either before triggering the upgrade or after the upgrade has successfully completed.
+
+#### Upgrade to Pravega 0.7 or above from a version below 0.7
+
+When upgrading the Pravega Cluster from any version below 0.7 to version 0.7 or above, there are a few configuration changes that must be made to Pravega manifest either with the upgrade request or prior to starting the upgrade. This is because from version 0.7 onwards, we no longer use Rocks DB for the Pravega Segmenstore. These configuration changes are needed to ensure that we have setup memory and other parameters appropriately for segmentstore streaming cache to work as expected.
+
+The additional changes that need to be made to the Pravega manifest are the following
+
+1. Ensure that the **memory limits** field within **Segmentstore Resources** is set to an appropriate value (let's call it the POD_MEM_LIMIT). The Segment Store cannot consume an amount of memory higher than this, otherwise we will get an Out Of Memory Killed error and the pod will be evicted.
+
+2. We need to distribute the pod's memory (M) between JVM Heap and Direct Memory. For instance, if POD_MEM_LIMIT=16GB then we can set 4GB for JVM and the rest for Direct Memory (12GB) i.e. POD_MEM_LIMIT (16GB) = JVM Heap (4GB) + Direct Memory (12GB).
+We need to ensure that the sum of JVM Heap and Direct Memory is not higher than the pod memory limit. In general, we can get JVM Heap fixed to 4GB and make the Direct Memory as the variable part.
+These two options can be configured through `segmentStoreJVMOptions: ["-Xmx4g", "-XX:MaxDirectMemorySize=12g"]`.
+
+3. From Pravega 0.7 onwards the memory is completely in memory, and it uses the Direct Memory part of it. But there are other things that use Direct Memory as well (like Netty). So the configuration of the cache should be 1GB or 2GB below the Direct Memory value provided. Otherwise, it will lead to problems.
```

Comment:
Please reword this part:
```
From Pravega 0.7 onwards the memory is completely in memory, and it uses the Direct Memory part of it. But there are other things that use Direct Memory as well (like Netty). So the configuration of the cache should be 1GB or 2GB below the Direct Memory value provided. Otherwise, it will lead to problems.
```
 _Memory is completely in memory_??? There is no need to mention those details. Stick to the configuration changes to be made..... the implementation details will be explained by pravega release notes.